src/pages/CompareCharts.tsx

```diff
@@ -1,64 +1,143 @@
-import React, { useEffect } from "react";
+import React, { useEffect, useState } from "react";
 import { withRouter, RouteComponentProps } from "react-router-dom";
+import * as QueryString from "query-string";
 import { useCookies } from "react-cookie";
 
-import { Typography, Button, Col } from "antd";
+import { Typography, Button, Col, Spin } from "antd";
 import Header from "../components/Header";
 import DimensionComparator from "../components/DimensionComparator";
 
 import "../styles/Preview.css";
 import "../styles/Footer.css";
 
-import charts from "../dummyData/charts";
-import courses from "../dummyData/courses";
+import { API_DOMAIN } from "../config";
+import { createDimension } from "../types/dimension";
+import { createCourse } from "../types/course";
+import { Chart, createStubChart } from "../types/chart";
 
 const CompareCharts: React.FC<RouteComponentProps> = (props) => {
   const [cookies] = useCookies(["accessToken"]);
+  const [courseName, setCourseName] = useState<string | undefined>(undefined);
+  const params = QueryString.parse(props.location.search);
 
-  // TODO: Need to change this to grab data from backend
-  const chartsToCompare = [charts[0], charts[1], charts[2]];
+  const chartsToCompareIDs = window.history.state?.state?.chartIDs;
+  const [chartsToCompare, setCharts] = useState<Chart[]>([]);
 
   useEffect(() => {
     if (!cookies["accessToken"]) {
       props.history.push("/Login");
     }
   }, [cookies]);
 
+  useEffect(() => {
+    fetchCharts();
+  }, [chartsToCompareIDs, cookies, params.courseID]);
+
+  const fetchCharts = async (): Promise<void> => {
+    await fetch(`${API_DOMAIN}course/${params.courseID}`, {
+      method: "GET",
+      headers: {
+        Authorization: `Bearer ${cookies["accessToken"]}`,
+        "Content-Type": "application/json",
+        Accept: "application/json",
+      },
+    })
+      .then((res) => res.json())
```

Comment:
can we please add some code here such that if res.status is not 200, then we send them to the dashboard? to keep it consistent, look in DisplayCards (function: fetchDimensions).